src/YesSql.Provider.SqlServer/SqlServerDialect.cs

```diff
@@ -157,9 +157,14 @@ public override void Concat(StringBuilder builder, params Action<StringBuilder>[
 
         public override List<string> GetDistinctOrderBySelectString(List<string> select, List<string> orderBy)
         {
-            foreach(var o in orderBy.Where(o => o.Trim() != "," && o.Trim() != "DESC" && o.Trim() != "ASC"))
+            // Sql Server requires all Ordered fields to be part of the select when DISTINCT is used
+            
+            foreach(var o in orderBy)
             {
-                if (!select.Contains(o))
+                var trimmed = o.Trim();
+                
+                // Each order segment can be a field name, or a punctuation, so we filter out the punctuations 
+                if (trimmed != "," && trimmed != "DESC" && trimmed != "ASC" && !select.Contains(o))
```

Comment:
It would be better if it's like
```csharp
private static readonly string[] _orderList = { ",", "ASC", "DESC" }
...
if (!_orderList.Contains(trimmed) && !select.Contains(o))
```

Reply:
I was also thinking of using a hashset in this case, but with 3 elements it's really equivalent to having a flat set of checks. So here it's really about code styling.